docs/architecture/adr-014-versioned-namespaces.md

```diff
@@ -0,0 +1,195 @@
+
+For example, consider the scenario where at mainnet launch blobs are layed out according to the existing non-interactive default rules. In this scenario, blobs always start at an index aligned with the `BlobMinSquareSize`. The only supported namespace ID is `0`. At some point in the future, if we introduce new non-interactive default rules (e.g. [celestia-app#1161](https://github.com/celestiaorg/celestia-app/pull/1161)), we may also expand the range of available namespaces to include namespaces that start with a leading `0` or `1` byte. Users may opt in to using the new non-interactive default rules by submitting PFB transactions with a namespace ID version of `1`.
+
+- When the namespace starts with `0`, all blobs in the namespace conform to the previous set of non-interactive default rules.
+- When a namespace starts with `1`, all blobs in the namespace conform to the new set of non-interactive default rules.
+
+```go
+optionA := []byte{
+  0,                      // namespace version
+  1, 2, 3, 4, 5, 6, 7, 8, // namespace ID
+  1,          // info byte (sequence start indicator = true)
+  0, 0, 0, 3, // sequence length
+  1, 2, 3, // blob data
+  // 0 padding until share is full
+ }
+```
+
+Pros
+
+- This proposal preserves backwards compatibility with roll-ups that want to continue using the previous set of non-interactive default rules.
+
+Cons
+
+- Old nodes won’t be able to parse the namespaces with a leading `1` byte so this is a breaking change that forces a hard-fork at an upgrade height.
+
+### Option B: `Namespace Version | Namespace Length | Namespace ID`
+
+Another approach is to prefix a namespace ID with the version and length of the namespace ID. This schema has the benefit that parsers can support the use case of changing from a namespace length of 8 bytes to 32 bytes without a version bump (and corresponding implementation change).
+
+| Field | Number of bytes | Description |
+| --- | --- | --- |
+| Namespace Version | 1 | the version of the namespace ID |
+| Namespace Length | 1 | the number of bytes that the namespace occupies encoded as a big-endian uint8 |
+| Namespace ID | * (based on the previous field) | namespace ID of the share |
+
+```go
+optionB := []byte{
+  0,                                                     // namespace version
+  16,                                                    // namespace length
+  1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, // namespace ID
+  1,          // info byte (sequence start indicator = true)
+  0, 0, 0, 3, // sequence length
+  1, 2, 3, // blob data
+  // 0 padding until share is full
+ }
+```
+
+Pros
+
+- Supports variable length namespaces without a namespace version bump
+
+Cons
+
+- 2 bytes of metadata per share
+
+Notes
+
+- Option B is implementable at a later time if we adopt Option A.
+- It isn't immediately clear how NMT can support variable length namespaces.
+
+### Option C: `Share Version | Namespace ID`
```

Comment:
I'm still for this option. (I'm aware we've already decided. I just haven't yet been convinced that there is a specific need for namespace versioning that isn't covered by share versioning and square versioning).

I'm also a big fan for putting the version number in a fixed indexed position (like the first or last byte) as opposed to the 9th byte or 17th byte

Reply:
> I'm also a big fan for putting the version number in a fixed indexed position (like the first or last byte) as opposed to the 9th byte or 17th byte

FWIW the accepted proposal has a namespace version at index 0 and a share version at index 33